### PR TITLE
Enhance CardDAV query report to return a selective list of vcard prop…

### LINKED ISSUE #889: Support for property filters in addressbook-query REPORT
Following https://tools.ietf.org/html/rfc6352#section-10.4 it is possible to limit the properties of the vcards to be returned by a addressbook-query.

So basically we should be capable to process a request like this 

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<C:addressbook-query xmlns:D="DAV:" xmlns:C="urn:ietf:params:xml:ns:carddav">
  <D:prop>
    <D:getetag/>
    <C:address-data>
      <C:prop name="VERSION"/>
      <C:prop name="UID"/>
      <C:prop name="NICKNAME"/>
      <C:prop name="EMAIL"/>
      <C:prop name="FN"/>
      <C:prop name="TEL"/>
    </C:address-data>
  </D:prop>
</C:addressbook-query>
```

@evert are you interested in accepting an PR on this?
## 

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/37992122-support-for-property-filters-in-addressbook-query-report?utm_campaign=plugin&utm_content=tracker%2F294006&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F294006&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…erties - fixes #889 

@evert

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
is that related to my changes?



````
1) Sabre\DAV\Auth\Backend\AbstractBasicTest::testRequireAuth
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'Basic realm="writing unittests on a saturday night"'
+'Basic realm="writing unittests on a saturday night", charset="UTF-8"'

/home/deepdiver/Development/ownCloud/sabre-dav/tests/Sabre/DAV/Auth/Backend/AbstractBasicTest.php:69

**Comment by staabm** [comment]
I somehow remember a PR which was merged lately to add the charset for the basic auth backend. I guess this is not related to your PR

**Comment by staabm** [comment]
Found it:
https://github.com/fruux/sabre-http/pull/70

Not sure why it fails here, maybe a dependency version problem

File: lib/CardDAV/Plugin.php [hunk 1]
```diff
@@ -846,14 +847,26 @@ protected function negotiateVCard($input, &$mimeType = null) {
      *
      * @param string|resource $data
      * @param string $target
+     * @param array $propertiesFilter
      * @return string
      */
-    protected function convertVCard($data, $target) {
+    protected function convertVCard($data, $target, $propertiesFilter = null) {
```
Comment (evert) [inline lib/CardDAV/Plugin.php]: Can this get can `array` typehint?

**Comment by DeepDiver1975** [comment]
@evert done

**Comment by evert** [comment]
Awesome feature @DeepDiver1975

**Comment by DeepDiver1975** [comment]
My pleasure.  What's your view on backporting this?

**Comment by evert** [comment]
Totally up for a backport! Doesn't seem like this would cause any sort of BC breaks. Happy to all the way back to 3.0 (which is the lowest supported version).

**Comment by DeepDiver1975** [comment]
here we go:
- 3.3: https://github.com/fruux/sabre-dav/pull/910
- 3.2: https://github.com/fruux/sabre-dav/pull/909
- 3.1: https://github.com/fruux/sabre-dav/pull/911

**Pull request merged**